ui/ui-components/client/client.ts

```diff
@@ -190,20 +195,100 @@ export class Client {
       }
 
       if (error.response && error.response.data && error.response.data.error) {
+        if (this.redirectOnAccessError(error.response.data.error)) {
+          if (options.errorHandler && error.response.data.error.message) {
+            options.errorHandler.set({
+              message: error.response.data.error.message,
+            });
+          }
+          return new Promise<Response>(() => {});
+        }
+
         const err =
           error.response?.data?.error?.message ??
           error.response?.data?.error ??
           error;
 
-        options.errorHandler.set({ message: err });
-        const newError = Error(err);
+        const newError = new Error(err);
         newError["code"] = error.response?.data?.error?.code;
+
         throw newError;
       } else {
-        options.errorHandler.set({ message: error });
         throw error;
       }
     }
+  }
+
+  public request = async <Response = any>(
```

Comment:
```
  public request = async <Response = unknown>(
```
If we did this it would force typed responses everywhere. Probably outside the scope of this PR, but worth doing.

Reply:
Yep, I gave it a try but definitely outside the scope of the PR. a lot of function are still not specifying the type.